vowpalwabbit/learner.h

```diff
@@ -895,6 +925,22 @@ struct common_learner_builder
     _learner->persist_metrics_fd.base = _learner->learn_fd.base;
     return *static_cast<FluentBuilderT*>(this);
   }
+
+  FluentBuilderT& set_copy(void (*fn_ptr)(DataT&, size_t& from, size_t& to))
+  {
+    _learner->copy_clear_fd.copy_notify_f = (multiconfig_data::fn_copy)fn_ptr;
+    _learner->copy_clear_fd.data = _learner->learn_fd.data;
+    _learner->copy_clear_fd.base = _learner->learn_fd.base;
+    return *static_cast<FluentBuilderT*>(this);
+  }
+
+  FluentBuilderT& set_reset_zero(void (*fn_ptr)(DataT&, size_t&))
```

Comment:
Is reset_zero the same as clear offset?

Reply:
a reduction can choose to call to clear offset on the weights, the point of this function is a way for an upper reduction like automl to signal a reduction below to reset anything that has to do with a given offset.

Reply:
I was mainly asking why the naming is different between the setter and how it is called:
`set_reset_zero` vs `clear_offset_based`